### ISSUE TITLE
Ability to customize frame fitting for different aspect ratios

### ISSUE THREAD

**Issue by hybridherbst** [google/model-viewer#3014]
### Description

Model-Viewer does a good job at presenting a model and a seamless poster for approximately-square content.
However, it seems it's currently not possible to adjust whether height- or width- (or a mix) fitting is used; e.g. when the browser window is expanded horizontally, the _height_ of the content stays the same, whereas in some cases it would be desired that the _width_ of the content stays the same (or that a 50/50 mix is used).

#### Live Demo
- open https://thor-and-the-serpent.glitch.me/ on deskop
- make the browser window approx. 2:1 aspect (e.g. 1000x500)
- click on the "Catapult" annotation
- note the composition
- make the browser window approx 1:2 aspect (e.g. 500x1000)
- note the (now broken) composition, this view doesn't make sense anymore

Ideally, this could be chosen - choosing for the entire model would be fine, the desired behaviour is very content-dependant. The alternative would be to adjust field of view dynamically depending on aspect which would interfere with camera navigation and everything else. 

An option would be to define an "authored aspect" for the model and then when changing aspect, the new one is either "fit inner" or "fit outer" or "something inbetween" to the authored aspect?

The current alternative (authoring everything in a portrait format) seems undesired since it leads to very far-away views on wide desktops and e.g. horizontal tablets that don't "look right".

**Authored Landscape View**
![chrome_i73lUEFsbp](https://user-images.githubusercontent.com/2693840/144475682-65679999-8e12-47b5-96d1-0db1276e7c84.png)

**Resulting Portrait View (not good)**
![image](https://user-images.githubusercontent.com/2693840/144475654-d9ce5ccc-09b2-4848-8344-c8be2d4c7753.png)

### Version
- model-viewer: v1.9.2

**Comment by elalish** [comment]
Yes, here it becomes clear that MV was primarily designed around the single-model  orbit use case. The default framing focuses  on height because  it basically fits the model  by revolving  it around it's up axis and ensuring all parts will  be  visible from all camera orbit angles. Also, the default view has to allow a single poster  image to be fit exactly over it  regardless of the  element's  aspect ratio. Keeping the  rest of the API sane during aspect ratio  changes resulted  in the  behavior you're seeing. 

It's possible it could be  improved, but it probably won't be easy given  the  constraints above. If you're interested in giving it a shot, I'll be happy to review a PR. I  was definitely hoping this could be avoided by "adjusting  the  fov dynamically during aspect ratio changes" outside of MV. What problems have you run into there?
